lucene/core.tests/src/test/org/apache/lucene/core/tests/TestMMap.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.lucene.core.tests;
+
+import org.apache.lucene.store.MMapDirectory;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestMMap {
+  @Test
+  public void testUnmapSupported() throws Exception {
+    Assert.assertTrue("Lucene loaded as a module.", MMapDirectory.class.getModule().isNamed());
```

Comment:
What is the message, if you comment out the `requires jdk.unsupported` in core module?

Sorry I am away from laptop device.

Reply:
```
org.apache.lucene.core.tests.TestMMap.txt, copied below:
   >     java.lang.AssertionError: Unmapping is not supported, because Lucene cannot read 'jdk.unsupported' module.
   >         at junit@4.13.1/org.junit.Assert.fail(Assert.java:89)
```

So, fine?

Reply:
Perfect. I will merge and add another assert.